sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -367,17 +372,265 @@ internal class DefaultExperimentClient internal constructor(
             .library("experiment-android-client/${BuildConfig.VERSION_NAME}")
             .build().merge(providedUser)
     }
+
+
+    private fun addContext(user: ExperimentUser?): ExperimentUser {
+        val providedUser = this.config.userProvider?.getUser()
+        val mergedUserProperties = (user?.userProperties ?: emptyMap()) + (providedUser?.userProperties ?: emptyMap())
+
+        return ExperimentUser.builder()
+            .library("experiment-android-client/$VERSION_NAME")
+            .merge(this.config.userProvider?.getUser()).merge(this.user)
+            .userProperties(mergedUserProperties)
+            .build()
+    }
+
+    private fun evaluate(flagKeys: Set<String>): Map<String, Variant> {
+        val user = addContext(this.user)
+        val flags = topologicalSort(this.flags.getAll(), flagKeys)
```

Comment:
This can throw an exception, catch and log the error, then return an empty map. I need to make this change in JS.